Adding a flexible infoBox

MatMath opened the pull request:
Adding a **InfoBox** to the UI components following the color theme.

### Feature: (see mdx file)
Color Scheme(4): subtle / strong / warning / alert
Size options(3): small / simple / descriptive
Actions: Nothing or Secondary &/or Primary Button
Coverage: 100%

<img width="322" alt="Screen Shot 2020-06-23 at 4 46 46 PM" src="https://user-images.githubusercontent.com/2585361/85460516-4f06d080-b571-11ea-9b19-6313561d44cb.png">
<img width="546" alt="Screen Shot 2020-06-23 at 4 46 36 PM" src="https://user-images.githubusercontent.com/2585361/85460519-4f9f6700-b571-11ea-8274-7d6bb9780930.png">
<img width="387" alt="Screen Shot 2020-06-23 at 4 47 19 PM" src="https://user-images.githubusercontent.com/2585361/85460511-4e6e3a00-b571-11ea-931f-f9ec69e66ff3.png">



Missing (future PR): The Warning & Alert severity should have different Button color following the theme, but the current Button only support light/dark theme so I felt it would be out of the scope of this PR to change the button theme while adding a new feature.

oclero commented:
Hey, looks like the generated docs auto-commit is missing.

oclero commented on src/components/info-box/types.ts:
```diff
@@ -0,0 +1,12 @@
+export const InfoBoxSeverity = {
+  ALERT: 'alert',
+  STRONG: 'strong',
+  SUBTLE: 'subtle',
+  WARNING: 'warning'
+} as const;
+
+export const InfoBoxSize = {
+  SMALL: 'small',
+  SIMPLE: 'simple',
+  DESCRIPTIVE: 'descriptive'
+} as const;
```
Could you follow the same convention as for the other components?

```ts
export type InfoBoxSeverity = 'alert' | 'strong' | 'subtle' | 'warning';

export type InfoBoxSize = 'small' | 'simple' | 'descriptive';
```

oclero commented on src/components/info-box/InfoBox.tsx:
```diff
@@ -0,0 +1,93 @@
+import { InfoBoxSeverity, InfoBoxSize } from './types';
+import styles from './InfoBox.module.scss';
+
+type InfoBoxSizeList = 'small' | 'simple' | 'descriptive'; // To display via mdx
+type InfoBoxSeverityList = 'alert' | 'strong' | 'subtle' | 'warning'; // To display via mdx
```
See my other comment about these types. No need to declare them with two different ways, it will be harder to maintain. Please prefer the convention we chose for the other components.

MatMath replied:
Hi @oclero 
The UI-Component does have 2 different ways, ex `<Button>` for list and `<Alert>` for enum typing. 

If we define the component as here `type "a" | "b" | ...` the Gatsby docs show properly 👍. But we lose the flexibility of typescript with `InfoBoxSeverity.WARNING` and we have to hardcode the value in the code logic itself. It was the closes I came to bridge the Docs & TS flexibility but I understand and I will remove the const typing.

oclero replied:
Yes I understand but we don't use enums in ui-componets, unfortunately. You still get autocompletion and typing: it's not as bad as hardcoded strings.

oclero commented on src/components/info-box/InfoBox.tsx:
```diff
@@ -0,0 +1,93 @@
+  };
+
+  const showActions = () => {
+    return (
```
Could you move this function out of the functional component, so we avoid a function creation every time?

MatMath commented:
> Hey, looks like the generated docs auto-commit is missing.

@oclero I saw a creeping commit few days ago `docs: update docz documentation` but that includes 120+ files so I removed it. I thought that would be done in a CI-CD chore on merge, not part of a commit since it would create merging conflict hell and render the review impossible to do. It is because most of the changes were made on filepath!?! 
ie: `"fullpath":"/Users/pducolin/repos/ui-components/src/components/sidenav/Sidenav.mdx",`
to: `"fullpath":"/Users/mlegault/github/ui-components/src/components/sidenav/Sidenav.mdx",`
All `filename Hash` also changed due to the `filepath`...

If I manually ran `yarn exec-docz-update` nothing gets added. I found how I can trick it to trigger but it is still `68 modified and 79 added`. I can add them but I think it should be part of the CI-CD instead.

sanatshah commented on src/components/info-box/InfoBox.tsx:
```diff
@@ -0,0 +1,104 @@
+          size='small'
+          theme={severity === 'strong' ? 'dark' : 'light'}
+          className={styles.actionsBtn}
+          onClick={() => secondary()}
```
Nitpick : You could just pass the function itselft -> onClick={secondary}

MatMath replied:
I wish but lint does not agree with me. :( 
![onClick is not the same type ](https://user-images.githubusercontent.com/2585361/85865210-8bb21200-b793-11ea-9d26-e2dca0733fc5.png)

sanatshah replied:
Oh weird, sometime lint has a mind of its own

sanatshah commented:
Your commits seem to be unverified.  Ran into a similar thing before, had to add another gpg key for github and update my local git config for this repo to use that key
